osquery/tables/system/darwin/disk_encryption.cpp

```diff
@@ -26,6 +28,62 @@ const std::string kDeviceNamePrefix = "/dev/";
 // kCoreStorageIsEncryptedKey is not publicly defined
 // or documented because CoreStorage is a private framework
 #define kCoreStorageIsEncryptedKey_ "CoreStorage Encrypted"
+#define kIODeviceTreeChosenPath_ "IODeviceTree:/chosen"
+
+Status genUnlockIdent(CFDataRef& uuid) {
+  auto chosen =
+      IORegistryEntryFromPath(kIOMasterPortDefault, kIODeviceTreeChosenPath_);
+  if (chosen == MACH_PORT_NULL) {
+    return Status(1, "Could not open IOKit DeviceTree");
+  }
+
+  CFMutableDictionaryRef properties;
+  auto kr = IORegistryEntryCreateCFProperties(
+      chosen, &properties, kCFAllocatorDefault, kNilOptions);
+  IOObjectRelease(chosen);
+
+  if (kr != KERN_SUCCESS) {
+    return Status(1, "Could not get IOKit chosen properties");
+  }
+
+  CFTypeRef unlock_ident;
+  if (CFDictionaryGetValueIfPresent(
+          properties, CFSTR("efilogin-unlock-ident"), &unlock_ident)) {
+    if (CFGetTypeID(unlock_ident) != CFDataGetTypeID()) {
+      return Status(1, "Unexpected data type for unlock ident");
+    }
+    uuid = (CFDataRef)unlock_ident;
+    return Status(0, "ok");
+  }
+
+  CFRelease(properties);
+  return Status(1, "Could not get unlock ident");
+}
+
+Status genUid(id_t& uid) {
+  CFDataRef uuid = nullptr;
+  if (!genUnlockIdent(uuid).ok()) {
+    return Status(1, "Could not get unlock ident");
+  }
+
+  char* buffer = new char[37];
+  CFDataGetBytes(uuid, CFRangeMake(0, CFDataGetLength(uuid)), (UInt8*)buffer);
```

Comment:
Will this scrub `buffer` to 0s? If not, perhaps a `memset` or just stack allocate with an initializer list and pass a reference to:
```
char *buffer[37] = {0};
```